src/Microsoft.ML.LightGBM/LightGbmBinaryTrainer.cs

```diff
@@ -0,0 +1,143 @@
+    {
+        public const string Summary = "LightGBM Binary Classifier";
+        public const string LoadNameValue = "LightGBMBinary";
+        public const string ShortName = "LightGBM";
+
+        public LightGbmBinaryTrainer(IHostEnvironment env, LightGbmArguments args)
+            : base(env, args, PredictionKind.BinaryClassification, "LGBBINCL")
+        {
+        }
+
+        public override IPredictorWithFeatureWeights<float> CreatePredictor()
+        {
+            Host.Check(TrainedEnsemble != null, "The predictor cannot be created before training is complete");
+            var innerArgs = LightGbmInterfaceUtils.JoinParameters(Options);
+            var pred = new LightGbmBinaryPredictor(Host, TrainedEnsemble, FeatureCount, innerArgs);
+            var cali = new PlattCalibrator(Host, -0.5, 0);
+            return new FeatureWeightsCalibratedPredictor(Host, pred, cali);
+        }
+
+        protected override void CheckDataValid(IChannel ch, RoleMappedData data)
+        {
+            Host.AssertValue(ch);
+            base.CheckDataValid(ch, data);
+            var labelType = data.Schema.Label.Type;
+            if (!(labelType.IsBool || labelType.IsKey || labelType == NumberType.R4))
+            {
+                throw ch.ExceptParam(nameof(data),
+                    $"Label column '{data.Schema.Label.Name}' is of type '{labelType}', but must be key, boolean or R4.");
+            }
+        }
+
+        protected override void CheckAndUpdateParametersBeforeTraining(IChannel ch, RoleMappedData data, float[] labels, int[] groups)
+        {
+            Options["objective"] = "binary";
+            // Add default metric.
+            if (!Options.ContainsKey("metric"))
+                Options["metric"] = "binary_logloss";
+        }
+    }
+
+    /// <summary>
+    /// A component to train an LightGBM model.
+    /// </summary>
+    public static partial class LightGbm
+    {
+        [TlcModule.EntryPoint(Name = "Trainers.LightGbmBinaryClassifier", Desc = "Train an LightGBM binary class model", UserName = LightGbmBinaryTrainer.Summary, ShortName = LightGbmBinaryTrainer.ShortName)]
+        public static CommonOutputs.BinaryClassificationOutput TrainBinary(IHostEnvironment env, LightGbmArguments input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register("TrainLightGBM");
+            host.CheckValue(input, nameof(input));
+            EntryPointUtils.CheckInputArgs(host, input);
+
+            return LearnerEntryPointsUtils.Train<LightGbmArguments, CommonOutputs.BinaryClassificationOutput>(host, input,
+                () => new LightGbmBinaryTrainer(host, input),
+                getLabel: () => LearnerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.LabelColumn),
+                getWeight: () => LearnerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.WeightColumn));
```

Comment:
>getWeight: () => LearnerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.WeightColumn) [](start = 16, length = 104)

LightGbmArguments is LearnerInputBaseWithGroupId, but we don't pass getGroup function...